src/deserialize/parse.rs

```diff
@@ -45,6 +46,21 @@ fn read_str(chars: &mut std::str::Chars) -> Edn {
     Edn::Str(string)
 }
 
+fn read_inst(chars: &mut std::str::Chars) -> Result<Edn, Error> {
+    let inst = chars
+        .take_while(|c| c != &'\"' || (*c).is_numeric())
+        .collect::<String>();
+    let time = chars.take_while(|c| c != &'\"').collect::<String>();
+
+    if inst.contains("inst") {
```

Comment:
Shoudn't this be more precise instead of `contains`? :thinking:

Reply:
just improved a little. What do you suggest? `!= "inst"`?

Reply:
If we use ` .take_while(|c| "inst ".contains(*c))` the error doesn't make sense:
```
 panic message: `"called `Result::unwrap()` on an `Err` value: ParseEdn(\"isdf 234 could not be parsed\")"`,
 expected substring: `"iasdf 234  could not be parsed"`
```
For test:
```
 #[test]
    #[should_panic(expected = "iasdf 234  could not be parsed")]
    fn panic_for_wrong_inst() {
        let mut edn = "#iasdf 234".chars();
        let a = parse(edn.next(), &mut edn);

        a.unwrap();
    }
```

Reply:
What I mean by precise is that I guess someone can pass something like `#instantenous "..."` or `#blablainst "..."`

Can't we do `starts_with` and then check that the rest is just extra spaces? :thinking:

Reply:
Ah OK! makes sense

Reply:
Will fix

Reply:
0f8f94d

Reply:
So there's still the problem with `instanteneous` for example. I've cloned the branch and created this test, but it should not pass, and it passes:

```rust
    #[test]
    fn parse_edn_with_instanteneous() {
        let mut edn =
            "@ :a :b {:c :d :date  #instanteneous \"2020-07-16T21:53:14.628-00:00\" ::c ::d} nil}".chars();

        assert_eq!(
            parse(edn.next(), &mut edn).unwrap(),
            Edn::Set(Set::new(set! {
                Edn::Key(":a".to_string()),
                Edn::Key(":b".to_string()),
                Edn::Map(Map::new(map! {
                    ":c".to_string() => Edn::Key(":d".to_string()),
                    ":date".to_string() => Edn::Inst("2020-07-16T21:53:14.628-00:00".to_string()),
                    "::c".to_string() => Edn::Key("::d".to_string())
                })),
                Edn::Nil
            }))
        )
    }
```

Sorry to not bring a solution, I am a bit busy.

Reply:
Okay then :confused: